modin/pandas/test/dataframe/test_protocol.py

```diff
@@ -0,0 +1,276 @@
+        #assert convert_column_to_array(df2.get_column_by_name(col_name) == df[col_name].tolist()
+        assert df[col_name].dtype == df2[col_name].dtype
+    
+    dict_null = {}
+    for col_name in df.columns:
+        num_null = np.random.randint(5)
+        list_null = []
+        num_nulls = 0
+        for ind in num_null:
+            ind_null = np.random.randint(5)
+            if ind_null not in list_null:
+                (df[col_name])[ind_null] = None
+                list_null.append(ind_null)
+                num_nulls += 1
+        dict_null[col_name] = num_nulls
+    
+    df2 = df.__dataframe__()
+
+    assert df2.get_column_by_name("x").null_count == dict_null["x"]
+    assert df2.get_column_by_name("y").null_count == dict_null["y"]
+    assert df2.get_column_by_name("z").null_count == dict_null["z"]
+
+
+def test_string():
+    df = pd.DataFrame({"A": ["a", None, "cdef", "", "g"]})
+    col = df.__dataframe__().get_column_by_name("A")
+
+    assert col.size == 5
+    assert col.null_count == 1
+    assert col.dtype[0] == DTypeKind.STRING
+    if StorageFormat.get() != "Omnisci":
+        assert col.describe_null == (4, 0)
+
+    df2 = df.__dataframe__()
+    assert df2.A.tolist() == df.A.tolist()
+    assert df2.get_column_by_name("A").null_count == 1
+    if StorageFormat.get() != "Omnisci":
+        assert df2.get_column_by_name("A").describe_null == (4, 0)
+    assert df2.get_column_by_name("A").dtype[0] == DTypeKind.STRING
+
+    df_sliced = df[1:]
+    col = df_sliced.__dataframe__().get_column_by_name("A")
+    assert col.size == 4
+    assert col.null_count == 1
+    assert col.dtype[0] == DTypeKind.STRING
+    if StorageFormat.get() != "Omnisci":
+        assert col.describe_null == (4, 0)
+
+    df2 = df_sliced.__dataframe__()
+    assert df2.A.tolist() == df_sliced.A.tolist()
+    assert df2.get_column_by_name("A").null_count == 1
+    if StorageFormat.get() != "Omnisci":
+        assert df2.get_column_by_name("A").describe_null == (4, 0)
+    assert df2.get_column_by_name("A").dtype[0] == DTypeKind.STRING
+
+
+def test_object():
+    df = pd.DataFrame({"x": np.array([None, True, False])})
+    col = df.__dataframe__().get_column_by_name("x")
+
+    assert col.size == 3
+
+    with pytest.raises(NotImplementedError):
+        assert col.dtype
+    with pytest.raises(NotImplementedError):
+        assert col.describe_null
+
+
+def test_DataFrame():
+    df = pd.DataFrame(
+        {
+            "x": np.array([True, True, False]),
+            "y": np.array([1, 2, 0]),
+            "z": np.array([9.2, 10.5, 11.8]),
+        }
+    )
+
+    df2 = df.__dataframe__()
+
+    assert df2._allow_copy is True
+    assert df2.num_columns() == 3
+    assert df2.num_rows() == 3
+    assert df2.num_chunks() == 1
+
+    assert list(df2.column_names()) == ["x", "y", "z"]
+
+    assert (
+        df2.select_columns((0, 2))._df[:, 0].tolist()
+        == df2.select_columns_by_name(("x", "z"))._df[:, 0].tolist()
```

Comment:
Please don't use private `._df` attribute and any variations of them (`._col`), these are implementation specific and differs for PandasOnRay and OmnisciOnNative implementations. 

We can test the same using `from_dataframe` function withouth accessing private properties of exchange objects.

Reply:
@dchigarev I changed this